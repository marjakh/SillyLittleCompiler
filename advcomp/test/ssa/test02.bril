@main {
  a: bool = const true;
  br a .true .false;
  .true:
  b: bool = const true;
  jmp .end;
  .false:
  b: int = const 0;
  jmp .end;
  .end:
  print b;
}
