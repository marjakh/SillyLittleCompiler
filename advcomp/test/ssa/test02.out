@main {
  a0: bool = const true;
  br a0 .true .false;
.true:
  b0: bool = const true;
  jmp .end;
.false:
  b1: int = const 0;
  jmp .end;
.end:
  b2 = phi b0 b1 .true .false;
  print b2;
}
